MSProject/ViewControllers/LibraryTableView/WBLibraryTableViewController.swift

```diff
@@ -116,5 +178,37 @@ extension WBLibraryTableViewController: UITableViewDelegate {
         let detailBookViewController = WBDetailBookViewController(with: book)
         navigationController?.pushViewController(detailBookViewController, animated: true)
     }
+}
+
+extension WBLibraryTableViewController: UISearchControllerDelegate {
+    
+    func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+        refreshControl = nil
+        _view.bookTable.refreshControl = nil
+    }
+    
+    func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        initRefreshControl()
+    }
+    
+    func didDismissSearchController(_ searchController: UISearchController) {
+        if #available(iOS 11.0, *) {
+            self.navigationItem.searchController = nil
+        } else {
+            self._view.bookTable.tableHeaderView = nil
+        }
+    }
+}
+
+extension WBLibraryTableViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, selectedScopeButtonIndexDidChange selectedScope: Int) {
+        filterContentForSearchText(searchBar.text!)
+    }
+}
+
+extension WBLibraryTableViewController: UISearchResultsUpdating {
+    func updateSearchResults(for searchController: UISearchController) {
+        filterContentForSearchText(searchController.searchBar.text!)
+    }
```

Comment:
looks pretty similar to ray wenderlich's tutorial on searchControllers :P. You can probably implement some of this with reactive and avoid using some of this delegates, but it's completely fine as it is.

Reply:
"pretty similar" you say? :p
thats why isFiltering() was there. lol